### PR TITLE
Re configure and annotate to use Jackson instead of Gson.

### PR DESCRIPTION
**Pull request by TjlHope**
So that it actually can work...

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2,175 +2,196 @@
...
+  </dependencies>
 
-    <scm>
-        <connection>git@bitbucket.org:mbst/atlas-deer-client.git</connection>
-        <developerConnection>git@bitbucket.org:mbst/atlas-deer-client.git</developerConnection>
-    </scm>
+  <scm>
+    <connection>git@bitbucket.org:mbst/atlas-deer-client.git</connection>
+    <developerConnection>git@bitbucket.org:mbst/atlas-deer-client.git</developerConnection>
+  </scm>
```
Comment (jamiesprax) [inline pom.xml]: change these to the github urls. We're not using bitbucket anymore

File: src/main/java/org/atlasapi/deer/client/AtlasClient.java [hunk 2]
```diff
@@ -20,10 +22,15 @@
     private final AtlasHttpClient httpClient;
     private final AtlasUrlCreator urlCreator;
 
-    public AtlasClient(HttpTransport httpTransport, String schema, HostSpecifier host,
-            String apiKey) {
+    public AtlasClient(
```
Comment (jamiesprax) [inline src/main/java/org/atlasapi/deer/client/AtlasClient.java]: 3 params is the max for constructors in our code style. Use a builder, pass it in, and then we'd only need the single constructor

File: src/main/java/org/atlasapi/deer/client/model/ContentResponse.java [hunk 3]
```diff
@@ -28,24 +31,33 @@ private ContentResponse(
...
         this.results = results;
         this.request = request;
     }
 
+    @JsonProperty
     public List<Content> getContent() {
         return content;
     }
 
+    @JsonIgnore   // TODO
```
Comment (jamiesprax) [inline src/main/java/org/atlasapi/deer/client/model/ContentResponse.java]: When we write TODO's we make a jira ticket with what there is to do and write the ticket number next to the todo. This usually just makes you do it as the effort of making the ticket could just be spent on the todo itself :)

Comment (TjlHope) [inline src/main/java/org/atlasapi/deer/client/model/ContentResponse.java]: That was just a reminder for me to check whether something more intelligent can be done.
But it'll do for now

**Pull request merged**